File: ros2-cpp-outputs/agr/src/static_map_server_node.cpp
// StaticMapServer — node implementation. EDIT ME.
//
// Generated ONCE by the ros2-sysmlv2 bridge pipeline and never overwritten
// (generation-gap pattern): this file is where demo/application logic lives.
// The architectural wiring is in static_map_server_node_base.hpp/.cpp, regenerated
// on every pipeline run.
#include "autonomous_ground_robot/static_map_server_node_base.hpp"

class StaticMapServer : public StaticMapServerBase
{
public:
  using StaticMapServerBase::StaticMapServerBase;

protected:
  // Override on_configure_hook / on_activate_hook / on_deactivate_hook here.
};

int main(int argc, char ** argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<StaticMapServer>();
  rclcpp::spin(node->get_node_base_interface());
  rclcpp::shutdown();
  return 0;
}

File: ros2-py-outputs/agr/autonomous_ground_robot/static_map_server_node.py
"""StaticMapServer — node implementation. EDIT ME.

Generated ONCE by the ros2-sysmlv2 bridge pipeline and never overwritten
(generation-gap pattern): this file is where demo/application logic lives.
The architectural wiring (endpoints, parameters, lifecycle transitions) is in
static_map_server_node_base.py, which is regenerated on every pipeline run.
"""
import rclpy

from .static_map_server_node_base import StaticMapServerBase


class StaticMapServer(StaticMapServerBase):
    """map_server application logic."""

    # Override on_configure_hook / on_activate_hook / on_deactivate_hook here.


def main(args=None):
    rclpy.init(args=args)
    node = StaticMapServer()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()
